c10/core/TensorImpl.cpp

```diff
@@ -88,6 +88,7 @@ TensorImpl::TensorImpl(Storage&& storage, DispatchKeySet key_set, const caffe2::
   if (c10::InferenceMode::is_enabled()) {
     // See Note [Expected TLS state in InferenceMode] for why we don't add Autograd & InplaceOrView keys.
     key_set_ = key_set;
+    version_counter_ = VariableVersion(-1);
```

Comment:
Don't do it this way. If you represent missing VC as -1, you still have to do the dynamic allocation for the variable version intrusive ptr target. Instead, represent not present version counter as the intrusive ptr nullptr equivalent.